pages/answers.tsx

```diff
@@ -56,70 +159,44 @@ const Answers: NextPage = () => {
   const [answers, setAnswers] = useState<any[]>([]);
   const [error, setError] = useState<string | null>(null);
 
-  const nameFingerprint = "73ZLcYHHAhx6rgMbgpPgc0F1qE89oWOAq6UGdrHkkJQ";
-  const allKeys = useE2ESDKClientKeys();
-  const currentKey = allKeys?.[nameFingerprint]?.[0] ?? null;
+  const identity = useE2ESDKClientIdentity();
 
-  console.log("allKeys", allKeys);
+  const fetchAnswers = useCallback(
+    () =>
+      fetchHasura({ query }, token).then((res: AnswersResponse) =>
+        res.data.answers
+          .map((answer) => decryptAnswer(client, answer))
+          .filter(Boolean)
+      ),
+    [client, token]
+  );
 
   useEffect(() => {
-    if (!currentKey) {
-      setError(
-        `Vous devez demander la permission pour accéder à "${nameFingerprint}"`
-      );
-      return;
-    }
-    if (!token.accessToken) {
-      setError(`You must be logged to access ${nameFingerprint}`);
-      return;
-    }
     setError(null);
-
-    fetchHasura({ query }, token).then((res: AnswersResponse) => {
-      const answers = res.data.answers
-        .map((answer) => {
-          try {
-            const values = client.unsealFormData(
-              {
-                metadata: {
-                  publicKey: answer.public_key,
-                  sealedSecret: answer.sealed_secret,
-                  signature: answer.signature,
-                },
-                encrypted: { data: answer.data },
-              },
-              currentKey.nameFingerprint
-            ) as Record<"data", string>;
-            const decryptedValues: DecryptedData = JSON.parse(values.data);
-            const res = jsonDataSchema.safeParse(decryptedValues);
-            if (!res.success) {
-              setError(`Zod: Impossible de parser la réponse ${answer.id}`);
-              console.error(res);
-              return null;
-            }
-            const decryptedAnswer = {
-              ...answer,
-              ...res.data,
-              data: undefined,
-            };
-            return [
-              decryptedAnswer.id,
-              decryptedAnswer.created_at.substring(0, 16),
-              decryptedAnswer.firstName,
-              decryptedAnswer.lastName,
-              decryptedAnswer.message,
-            ];
-          } catch (e) {
-            console.error(e);
-            setError(`e2esdk: Impossible de parser la réponse ${answer.id}`);
-            return null;
+    if (session?.user?.email) {
+      const e2esdkUserId = session.user.email;
```

Comment:
**suggestion**: I know that this may be an example application, but a friendly reminder that people change their emails, and the e2esdk user ID should be immutable.

Reply:
fixed